forch/simple_auth_state_machine.py

```diff
@@ -113,10 +129,16 @@ def handle_sm_timer(self):
                 backoff_time = backoff * self._query_timeout_sec
                 self._current_timeout = time.time() + backoff_time
                 if self._current_state == self.REQUEST:
+                    if self._retry_backoff:
```

Comment:
I think we need something more explicit here (since this is about security).  From this implementation, it's somewhat obtuse when it will transition to the UNAUTH state.  Looking at this code I can't easily determine what the algorithm is.   Also, it's not clear how this maps to the comments above.  Do you have a clear description of what the behavior *should* be (independent of the implementation)?  AUTH -> AUTH and UNAUTH -> UNAUTH, and UNAUTH -> AUTH are all clear and obvious, but the AUTH -> UNAUTH process is unclear (where in the comment does it describe this?)

Reply:
Updated the comments.

For AUTH -> UNAUTH transition, it happens in `host_expired` when a device is disconnected from the switch. It will not be triggered by any timeout.

Reply:
That's not right -- eventually the device needs to be unauthenticated if it doesn't receive a valid radius response.

Reply:
Updated the timeout handling logics and added "REQUEST -> UNAUTH' transition. If the Forch does not receive an accepted radius response after `max_backoff` retries, the device will be set to UNAUTH state and will be deauthenticated. Once a device enters UNAUTH state, Forch will try to reauthenticate it again after `_rej_timeout_sec`.